pkg/pillar/cmd/zedrouter/metadata_handlers.go

```diff
@@ -650,7 +650,7 @@ func HandlePatchDownload(z *zedrouter) func(http.ResponseWriter, *http.Request)
 func HandlePatchFileDownload(z *zedrouter) func(http.ResponseWriter, *http.Request) {
```

Comment:
Is HandlePatchFileDownload also used for external binary blobs? If so, how does it work? Is GetZipArchive also used here or is the blob served by metadata server directly from the filepath where it is stored by volumemgr?

Reply:
It's served directly from the filepath where it is stored by volumemgr, when archive is created in temp directory and iterates through binary artifacts which are completed (when we receive VolumeStatus which indicates that file has been downloaded and is stored, we move it from VolumeRef to BinaryBlob see updateExternalPatches in zedrouter/patchenvelopes.go). In case of external binary blobs we will access file directly stored by volumemgr

Reply:
OK, now it makes sense to me, thanks,